libraries/ms-drivers/src/adt_7470_fan_controller.c

```diff
@@ -0,0 +1,61 @@
+#include "adt_7470_fan_controller.h"
+
+#include <stddef.h>
+
+#include "adt_7470_fan_controller_defs.h"
+
+#define STM32_GPIO_STATE_SELECT_OUT_0 GPIO_STATE_LOW
+#define STM32_GPIO_STATE_SELECT_OUT_1 GPIO_STATE_HIGH
+// need to set interrupt once fan goes out of range
+// PWM duty cycle is set from 0-100, in steps of 0.39 (0 - 0xFF)
+StatusCode apt7470_set_speed(I2CPort port, uint8_t *speed, Fan FAN, uint16_t ADR7470_I2C_ADDRESS) {
+  // need to select duty fan - correct fan duty cycle
+  return i2c_write_reg(I2C_PORT_1, ADR7470_I2C_ADDRESS, ADT7470_PWM2_DUTY_CYCLE, speed,
+                       SET_SPEED_NUM_BYTES);
+}
+
+static StatusCode prv_init_common(Adt7470Storage *storage) {
+  // i2c pin stuff goes here
+  I2CSettings i2c1_settings = {
+    .speed = I2C_SPEED_FAST,
+    .sda = I2C1_SDA,
+    .scl = I2C1_SCL,
+  };
+
+  i2c_init(I2C_PORT_1, &i2c1_settings);
+
+  return STATUS_CODE_OK;
+}
+
+StatusCode bts_7240_init(Adt7470Storage *storage, Adt7470Settings *settings) {
```

Comment:
bts 7240 ?? I swear you're just using random letters and numbers at this point 😅